packages/flutter/lib/src/cupertino/theme.dart

```diff
@@ -0,0 +1,314 @@
+  /// A color used for content that must contrast against a [primaryColor] background.
+  ///
+  /// For example, this color is used for a [CupertinoButton]'s text and icons
+  /// when the button's background is [primaryColor].
+  ///
+  /// If coming from a Material [Theme] and unspecified, [primaryContrastingColor]
+  /// will be derived from the Material [ThemeData]'s `colorScheme.onPrimary`.
+  ///
+  /// Reading this property from [CupertinoTheme.of] will create a dependency
+  /// from the [BuildContext] to changes in this property.
+  Color get primaryContrastingColor {
+    return _primaryContrastingColor ??
+        (_isLight ? CupertinoColors.white : CupertinoColors.black);
+  }
+
+  final CupertinoTextTheme _textTheme;
+  /// Text styles used by Cupertino widgets.
+  ///
+  /// Derived from [brightness] and [primaryColor] if unspecified, including
+  /// [brightness] and [primaryColor] of a Material [ThemeData] if coming
+  /// from a Material [Theme].
+  ///
+  /// Reading this property from [CupertinoTheme.of] will create a dependency
+  /// from the [BuildContext] to changes in this property.
+  CupertinoTextTheme get textTheme {
+    return _textTheme ?? CupertinoTextTheme(
+      brightness: brightness,
+      primaryColor: primaryColor,
+    );
+  }
+
+  final Color _barBackgroundColor;
+  /// Background color of the top nav bar and bottom tab bar.
+  ///
+  /// Defaults to a light gray or a dark gray translucent color depending
+  /// on the [brightness].
+  ///
+  /// Reading this property from [CupertinoTheme.of] will create a dependency
+  /// from the [BuildContext] to changes in this property.
+  Color get barBackgroundColor {
+    return _barBackgroundColor ??
+        (_isLight ? _kDefaultBarLightBackgroundColor : _kDefaultBarDarkBackgroundColor);
+  }
+
+  final Color _scaffoldBackgroundColor;
+  /// Background color of the scaffold.
+  ///
+  /// Defaults to white or black depending on the [brightness].
+  ///
+  /// Reading this property from [CupertinoTheme.of] will create a dependency
+  /// from the [BuildContext] to changes in this property.
+  Color get scaffoldBackgroundColor {
+    return _scaffoldBackgroundColor ??
+        (_isLight ? CupertinoColors.white : CupertinoColors.black);
+  }
+
+  final Color _tableBackgroundColor;
+  /// Background color of a table view behind cell groups.
+  ///
+  /// Defaults to a [CupertinoColors.extraLightBackgroundGray] or
+  /// [CupertinoColors.darkBackgroundGray] depending on the [brightness].
+  ///
+  /// Reading this property from [CupertinoTheme.of] will create a dependency
+  /// from the [BuildContext] to changes in this property.
+  Color get tableBackgroundColor {
+    return _tableBackgroundColor ??
+        (_isLight ? CupertinoColors.extraLightBackgroundGray : CupertinoColors.darkBackgroundGray);
+  }
+
+  /// Return an instance of the [CupertinoThemeData] whose property getters
+  /// only return the construction time specifications with no derived values.
```

Comment:
this explanation really doesn't help anyone who doesn't understand the subtleties of the class hierarchy here.

Reply:
added more text